### PR TITLE
Fix Fortran errors in gfortran-10

### LINKED ISSUE #1017: Compilation problem with gcc-10
On Debian, we are currently testing to build our packages with the new gcc release 10. For casacore, this leads to an errror as reported in [Debian#957065](https://bugs.debian.org/957065):

```
[ 50%] Building Fortran object scimath_f/CMakeFiles/casa_scimath_f.dir/dfftpak.f.o
cd /<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/scimath_f && /usr/bin/gfortran -DCFITSIO_VERSION_MAJOR=3 -DCFITSIO_VERSION_MINOR=470 -DHAVE_FFTW3 -DHAVE_FFTW3_THREADS -DHAVE_O_DIRECT -DHAVE_READLINE -DUSE_THREADS -DWCSLIB_VERSION_MAJOR=7 -DWCSLIB_VERSION_MINOR=1 -Dcasa_scimath_f_EXPORTS -I/<<PKGBUILDDIR>> -I/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu -I/usr/include  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -fPIC   -c /<<PKGBUILDDIR>>/scimath_f/dfftpak.f -o CMakeFiles/casa_scimath_f.dir/dfftpak.f.o
/<<PKGBUILDDIR>>/scimath_f/dfftpak.f:732:72:

  732 |       CALL DRFTB1 (N,R,WSAVE,WSAVE(N+1),WSAVE(2*N+1))
      |                                                                        1
Error: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4)
/<<PKGBUILDDIR>>/scimath_f/dfftpak.f:214:72:

  214 |       CALL DRFTF1 (N,R,WSAVE,WSAVE(N+1),WSAVE(2*N+1))
      |                                                                        1
Error: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4)
/<<PKGBUILDDIR>>/scimath_f/dfftpak.f:82:72:

   82 |       CALL DRFTI1 (N,WSAVE(N+1),WSAVE(2*N+1))
      |                                                                        1
Error: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4)
/<<PKGBUILDDIR>>/scimath_f/dfftpak.f:1284:72:

 1284 |       CALL DEFFT1 (N,WSAVE(2*N+1),WSAVE(3*N+1))
      |                                                                        1
Error: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4)
/<<PKGBUILDDIR>>/scimath_f/dfftpak.f:2213:72:

 2213 |       CALL DCFTI1 (N,WSAVE(IW1),WSAVE(IW2))
      |                                                                        1
Error: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4)
/<<PKGBUILDDIR>>/scimath_f/dfftpak.f:2341:72:

 2341 |       CALL DCFTF1 (N,C,WSAVE,WSAVE(IW1),WSAVE(IW2))
      |                                                                        1
Error: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4)
/<<PKGBUILDDIR>>/scimath_f/dfftpak.f:2806:72:

 2806 |       CALL DCFTB1 (N,C,WSAVE,WSAVE(IW1),WSAVE(IW2))
      |                                                                        1
Error: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4)
make[3]: *** [scimath_f/CMakeFiles/casa_scimath_f.dir/build.make:131: scimath_f/CMakeFiles/casa_scimath_f.dir/dfftpak.f.o] Error 1
make[3]: Leaving directory '/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu'
make[2]: *** [CMakeFiles/Makefile2:10230: scimath_f/CMakeFiles/casa_scimath_f.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
```
[Full build log](http://people.debian.org/~doko/logs/gcc10-20200225/casacore_3.2.1-4_unstable_gcc10.log)

Common problems with the new version are discussed [here](http://gcc.gnu.org/gcc-10/porting_to.html#fortran).

### PR DESCRIPTION
**Pull request by aroffringa**
This PR is supposed to fix #1017 . gfortran 10 has changed some behaviour, and some of the things that were previously warnings are now errors and thus cause compilation to fail.

### CODE REVIEW

**Comment by aroffringa** [comment]
This branch now succesfully compiles with gfortran 10 (and gcc/g++ 10). It would be good if someone reviews the changes to the Fortran code (`lawson.f`); I went through them twice but it is easy to make a mistake in the line nrs. Tests run fine.

File: scimath/CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ Functionals/FunctionFactoryErrors.cc
 Functionals/FunctionalProxy.cc
 Functionals/SerialHelper.cc
 Mathematics/Combinatorics.cc
-Mathematics/FFTPack.cc
+#Mathematics/FFTPack.cc
```
Comment (tammojan) [inline scimath/CMakeLists.txt:17]: To be deleted (commenting out will likely only confuse since the file is gone).

File: scimath_f/lawson.f [hunk 2]
```diff
@@ -993,15 +1051,16 @@ SUBROUTINE DNNLS (A,MDA,M,N,B,X,RNORM,W,ZZ,INDEX,ITMAX,MODE)
 C
       IF (JJ.EQ.NSETP) GO TO 290
       JJ=JJ+1
-          DO 280 J=JJ,NSETP
+        DO J=JJ,NSETP
           II=INDEX(J)
           INDEX(J-1)=II
           CALL DG1 (A(J-1,II),A(J,II),CC,SS,A(J-1,II))
           A(J,II)=ZERO
               DO 270 L=1,N
```
Comment (tammojan) [inline scimath_f/lawson.f:1059]: Why did you not replace this with a line number less `DO`? I.e. replace this with
```
DO L=1,N
  IF (L.NE.II) CALL DG2 (CC,SS,A(J-1,L),A(J,L))
END DO
```

File: scimath_f/lawson.f [hunk 3]
```diff
@@ -855,20 +905,22 @@ SUBROUTINE DNNLS (A,MDA,M,N,B,X,RNORM,W,ZZ,INDEX,ITMAX,MODE)
...
 C                                   FIND LARGEST POSITIVE W(J).
    60 WMAX=ZERO
-          DO 70 IZ=IZ1,IZ2
+        DO 70 IZ=IZ1,IZ2
           J=INDEX(IZ)
           IF (W(J).LE.WMAX) GO TO 70
           WMAX=W(J)
           IZMAX=IZ
-   70     CONTINUE
+   70   CONTINUE
```
Comment (tammojan) [inline scimath_f/lawson.f:923]: I think this should be `END DO`. The `CONTINUE` statement is a no-op.

File: scimath_f/lawson.f [hunk 4]
```diff
@@ -824,6 +872,7 @@ SUBROUTINE DLDP (G,MDG,M,N,H,X,XNORM,W,INDEX,MODE)
       SUBROUTINE DNNLS (A,MDA,M,N,B,X,RNORM,W,ZZ,INDEX,ITMAX,MODE)
       implicit double precision (a-h,o-z)
       DIMENSION A(MDA,N), B(1), X(1), W(1), ZZ(1)
+      DOUBLE PRECISION DUMMY(1)
```
Comment (tammojan) [inline scimath_f/lawson.f:875]: Why is this here?

File: scimath_f/lawson.f [hunk 5]
```diff
@@ -290,7 +310,7 @@ SUBROUTINE NNLS (A,MDA,M,N,B,X,RNORM,W,ZZ,INDEX,ITMAX,MODE)
       DIMENSION A(MDA,N), B(1), X(1), W(1), ZZ(1)
       INTEGER INDEX(1)
       REAL      A, B, X, RNORM, W, ZZ, ZERO, ONE, TWO, FACTOR, SM,
-     $   WMAX, ASAVE, UP, DUMMY, UNORM, ZTEST, ALPHA, T, CC, SS
+     $   WMAX, ASAVE, UP, DUMMY(1), UNORM, ZTEST, ALPHA, T, CC, SS
```
Comment (tammojan) [inline scimath_f/lawson.f:313]: If this is the same dummy that is declared as double precision below, it is contradictory.

File: scimath_f/lawson.f (hunk 4 above)
Comment (aroffringa) [reply scimath_f/lawson.f:875]: It's required to define to give it 'the right shape'. If I remember well: Old Fortran allowed undefined use of Dummy, and made it a scalar. Old fortran than accepted that scalar in a function call and convert it to a one-dim array.

File: scimath_f/lawson.f (hunk 3 above)
Comment (tammojan) [reply scimath_f/lawson.f:923]: Ah wait, you did not convert this DO-loop to the line-number free variant. So it should be fine.

File: scimath_f/lawson.f (hunk 2 above)
Comment (tammojan) [reply scimath_f/lawson.f:1059]: As discussed offline, not all DO's need to be replaced with the `DO` / `END DO` form.

File: scimath_f/lawson.f (hunk 5 above)
Comment (aroffringa) [reply scimath_f/lawson.f:313]: Aren't they in different subroutines?

Comment (tammojan) [reply scimath_f/lawson.f:313]: Ah it's in different functions, so is fine.

File: scimath/CMakeLists.txt (hunk 1 above)
Comment (aroffringa) [reply scimath/CMakeLists.txt]: indeed; removed in new commit

**Comment by tammojan** [comment]
Could you remove remaining references to FFTPack? In particular the include in Mathematics.h, that crashes the build.
```
./scimath/CMakeLists.txt:#Mathematics/FFTPack.cc
./scimath/CMakeLists.txt:Mathematics/FFTPack.h
./scimath/Mathematics/FFTServer.h:  // FFTW specific members. Do not harm if FFTPack is used.
./scimath/Mathematics/FFTServer.hcc://# or FFTPack.
./scimath/Mathematics/FFTServer.hcc://# about using FFTW3 or FFTPack is only needed when building casacore).
./scimath/Mathematics.h:#include <casacore/scimath/Mathematics/FFTPack.h>
./scimath/Mathematics.h://      the functions in the <linkto class="FFTPack">FFTPack</linkto>
./scimath/Mathematics.h://      class. The FFTPack class also contains functions for doing
```

**Comment by tammojan** [comment]
Also the FFTPack-specific member `itsWork` of FFTServer can go, along with the `delete` statements in the `FFTServer` destructors.
